python/ray/air/checkpoint.py

```diff
@@ -758,6 +800,11 @@ def __fspath__(self):
             "Use `Checkpoint.to_directory()` or `Checkpoint.as_directory()` instead."
         )
 
+    def __del__(self):
+        if self._managed:
+            assert self._local_path
```

Comment:
For a checkpoint created via this flow, should we hide the public `uri`?

Reply:
Good point, I'll add that to the PR